app/models/data.py

```diff
@@ -242,10 +242,38 @@ def totalTestResults(self):
     def parse_str_to_date(date_input):
         return parser.parse(str(date_input), ignoretz=True).date()
 
-    def __init__(self, **kwargs):
-        # strip any empty string fields from kwargs
-        kwargs = {k: v for k, v in kwargs.items() if v is not None and v != ""}
+    def field_diffs(self, dict_other):
+        ''' Return the list of fields that dict_other would modify if applied
+        on this row.
+        Some business logic is applied, and some fields are skipped from comparison, field
+        aliases get special treatment.
+
+        Return: List[str], empty list if the dict results in no changes to current data
+        '''
+        fields = []
+        if not dict_other:
+            return fields
+
+        # we want to compare after all parsing is done
+        other = CoreData(**dict_other)
+
+        # special casing for aliasse (maybe move elsewhere?)
+        if 'lastUpdateIsoUtc' in dict_other:
+            # the value is compared through the object, but the key should exist
+            dict_other['lastUpdateTime'] = dict_other['lastUpdateIsoUtc']
+
+        for field in CoreData.__table__.columns.keys():
+            # we expect batch IDs to be different, skip comparing those
+            if field == 'batchId':
+                continue
+            # for any other field, compare away
+            if field in dict_other and getattr(other, field) != getattr(self, field):
+                fields.append(field)
+
+        return fields
 
+    @staticmethod
+    def _cleanup_kwargs(kwargs):
```

Comment:
nit: could you rename to `_cleanup_date_kwargs` so we know this function is relatively date-specific? Kind of hoping we don't do any other kwarg cleanup anyway